packages/ui-components/src/components/NatPurseSelector.jsx

```diff
@@ -0,0 +1,85 @@
+// import React from 'react';
+// import { MenuItem, TextField } from '@material-ui/core';
+import clsx from 'clsx';
+// import { makeStyles } from '@material-ui/core/styles';
+import { MathKind } from '@agoric/ertp';
+
+import makeNatPurseListItemDetailed from './NatPurseListItemDetailed';
+
+const makeNatPurseSelector = (
+  React,
+  MenuItem,
+  TextField,
+  ListItemIcon,
+  ListItemText,
+  PurseIcon,
+  makeStyles,
+) => ({
+  purses: pursesUnfiltered,
+  purseSelected,
+  onChange,
+  disabled,
+  error,
+}) => {
+  const ListItem = makeNatPurseListItemDetailed(
+    React,
+    MenuItem,
+    ListItemIcon,
+    ListItemText,
+    PurseIcon,
+    makeStyles,
+  );
+
+  const useStyles = makeStyles(() => ({
+    select: {
+      display: 'flex',
+      alignItems: 'center',
+    },
+    noPadding: {
+      paddingTop: 0,
+      paddingBottom: 0,
+    },
+  }));
+
+  const classes = useStyles();
+
+  const noPurses = (
+    <MenuItem key={null} value={null}>
+      No purses
+    </MenuItem>
+  );
+
+  let items = noPurses;
+  if (pursesUnfiltered === null) {
+    pursesUnfiltered = [];
+  }
+
+  const isNatPurse = ({ displayInfo: { amountMathKind } }) =>
+    amountMathKind === MathKind.NAT;
+
+  const purses = pursesUnfiltered.filter(isNatPurse);
+  if (purses && purses.length > 0) {
+    items = purses.map(purse => (
+      <ListItem key={purse.pursePetname} purse={purse} />
+    ));
+  }
+  return (
+    <TextField
+      select
+      label="Currency"
+      variant="outlined"
+      fullWidth
+      value={purseSelected === null ? '' : purseSelected.pursePetname}
+      onChange={onChange}
+      inputProps={{
+        className: clsx(purseSelected && classes.noPadding, classes.select),
+      }}
+      disabled={disabled}
+      error={error}
+    >
+      {items}
+    </TextField>
+  );
+};
+
+export default makeNatPurseSelector;
```

Comment:
why is this exporting a `make` function rather than just the component?

Reply:
It's necessary so that there are not two copies of React and MaterialUI. If we were importing from npm, we wouldn't need to do this, but since we are linking locally, it's imported twice unless we pass it in like this.